Source/WebCore/rendering/updating/RenderTreeBuilderList.cpp

```diff
@@ -80,9 +80,16 @@ static std::pair<RenderBlock*, RenderBlock*> findParentOfEmptyOrFirstLineBox(Ren
     return { { }, fallbackParent };
 }
 
-static RenderBlock* parentCandidateForMarker(RenderBlock& blockContainer, const RenderListMarker& marker)
+static RenderBlock* parentCandidateForMarker(RenderListItem& listItemRenderer, const RenderListMarker& marker)
 {
-    auto [parentCandidate, fallbackParent ] = findParentOfEmptyOrFirstLineBox(blockContainer, marker);
+    auto* listItemParent = listItemRenderer.parent()->element();
+    if (marker.isInside() && listItemParent && isHTMLListElement(*listItemParent)) {
```

Comment:
I'm surprised this is specific to HTML `<li>` elements, and not simply to `display: list-item` (which is implied here).

@Loirooriol @fantasai Can you double check if this is correct?

Reply:
I think you just be able to drop ` && listItemParent && isHTMLListElement(*listItemParent)` in that case. I think it's implied that the parent is a list item in this context

Reply:
I don't understand @AlexsanderDamaceno's comment above, but I agree the logic shouldn't depend on whether the list item is a `<li>` or not.